reference.md

```diff
@@ -102,8 +102,9 @@ for loop
     See also: [while loop](#while-loop).
 
 function
-:   A group of instructions (i.e., lines of code) that transform
-    some input [arguments](#argument) to some output.
+:   A group of instructions (i.e., lines of code) that may be executed repeatedly,
+    every time it is [called](#function call). Functions can accept some input
+    [arguments](#argument), and they can [return](#return-statement) some output.
```

Comment:
How about:

> A named group of instructions that is executed when function's name is used in the code. Occurrence of a function name in the code is called "function call". Functions may process input arguments and return some value(s), or may be used to group together logical pieces of code without returning any value.

I'm not super happy about "without returning any value" in my suggestion above (function will return `None`), but it might be OK in our case.